Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -15,7 +16,176 @@ public extension PBXProj.Objects {
+    }
+
+    /// Returns group with the given name contained in the given parent group and its reference.
+    ///
+    /// - Parameter groupName: group name.
+    /// - Parameter inGroup: parent group.
+    /// - Returns: group with the given name contained in the given parent group and its reference.
+    public func group(named groupName: String, inGroup: PBXGroup) -> (String, PBXGroup)? {
+        let children = inGroup.children
+        return groups.first(where: {
+            children.contains($0.key) && ($0.value.name == groupName || $0.value.path == groupName)
+        })
+    }
+
+    /// Adds new group with the give name to the given parent group.
+    /// Group name can be a path with components separated by `/`.
+    /// Will create new groups for intermediate path components or use existing groups.
+    /// Returns all new or existing groups in the path and their references.
+    ///
+    /// - Parameters:
+    ///   - groupName: group name, can be a path with components separated by `/`
+    ///   - toGroup: parent group
+    ///   - options: additional options, default is empty set.
+    /// - Returns: all new or existing groups in the path and their references.
+    public func addGroup(named groupName: String, to toGroup: PBXGroup, options: GroupAddingOptions = []) -> [(String, PBXGroup)] {
+        return addGroups(groupName.components(separatedBy: "/"), to: toGroup, options: options)
+    }
+
+    private func addGroups(_ groupNames: [String], to toGroup: PBXGroup, options: GroupAddingOptions) -> [(String, PBXGroup)] {
+        guard !groupNames.isEmpty else { return [] }
+        let newGroup = createOrGetGroup(named: groupNames[0], in: toGroup, options: options)
+        return [newGroup] + addGroups(Array(groupNames.dropFirst()), to: newGroup.1, options: options)
+    }
+
+    private func createOrGetGroup(named groupName: String, in parentGroup: PBXGroup, options: GroupAddingOptions) -> (String, PBXGroup) {
+        if let existingGroup = group(named: groupName, inGroup: parentGroup) {
+            return existingGroup
+        }
+
+        let newGroup = PBXGroup(
+            children: [],
+            sourceTree: .group,
+            name: groupName,
+            path: options.contains(.withoutFolder) ? nil : groupName
+        )
+        let reference = generateReference(newGroup, groupName)
+        addObject(newGroup, reference: reference)
+        parentGroup.children.append(reference)
+        return (reference, newGroup)
+    }
+
+    /// Adds file at the give path to the project or returns existing file and its reference.
+    ///
+    /// - Parameters:
+    ///   - filePath: path to the file
+    ///   - sourceTree: file source tree, default is `.group`.
+    ///   - name: file name, by default gets file name from the path
+    /// - Returns: new or existing file and its reference.
+    public func addFile(at filePath: Path, sourceRoot: Path) throws -> (String, PBXFileReference) {
```

Comment:
Please add unit test for this function. At the moment, I have a feeling that `filePath == fullPath(fileElement: $0.value, reference: $0.key, sourceRoot: sourceRoot) ` check will fail unless `filePath` is initialized as absolute path. But it should not be a absolute path as `PBXFileReference ` is created with `sourceTree: .group `